### PR TITLE
Support options with only a short flag

### PR DESCRIPTION
**Pull request by shadowspawn**
# Pull Request

## Problem

Users may wish to declare an option which has only a short flag and no long flag.

Defining just a short flag for an option appeared to almost work, but was not actually supported internally.

See: #1249

## Solution

Extend option flag parsing to allow short, or long, or both. Extend the support to the custom option flags for `.helpOption()` and `.version()` too.

## ChangeLog

- allow options to be added with just a short flag (#1256)

### CODE REVIEW

**Comment by shadowspawn** [comment]
Side note: I hit a merge conflict when updating the branch which prompted me to look at how `negate` was set in the option constructor. I tightened the test to match the README and expectation in other code that it is for a long option that `startsWith` `--no-`.

**Comment by shadowspawn** [comment]
I created a `release/6.x` branch since this is a semver major change, although hopefully only breaking for people who were using lone short flags and working around the bugs.

I am thinking we might do a pre-release in June for v6, with short flags and variadic options.

File: index.js [hunk 1]
```diff
@@ -1708,6 +1711,28 @@ function humanReadableArgName(arg) {
...
+ *
+ * @api private
+ */
+
+function _parseOptionFlags(flags) {
+  let shortFlag;
+  let longFlag;
+  // Use original very loose parsing to maintain backwards compatibility for now,
+  // which allowed for example unintended `-sw, --short-word` [sic].
+  const flagParts = flags.split(/[ |,]+/);
```
Comment (glensc) [inline index.js:1725]: this pattern seems buggy if the comment is still accurate, did you intend to say `/( |,)+/` there? or just `/[ ,]+/` to match space or comma, or really wanted to add a pipe `|` to the mix?

Comment (shadowspawn) [reply index.js:1725]: The pipe is really in the mix. From the README:

> Each option can have a short flag (single character) and a long name, separated by a comma or space or vertical bar ('|').

The comment is about `-sw` which is neither a short flag nor a long name.

**Pull request merged**